### PR TITLE
[patch] Not create tmp file when copying everything

### PR DESCRIPTION
**Pull request by samwaseda**
@pmrv pointed this case out, and somehow the PR got closed so I open another one.

### CODE REVIEW

File: pyiron_base/project/archiving/export_archive.py [hunk 1]
```diff
@@ -38,7 +38,11 @@ def update_project(project_instance, directory_to_transfer, archive_directory, d
 
 
 def copy_files_to_archive(
-    directory_to_transfer, archive_directory, compress=True, copy_all_files=False, arcname=None
+    directory_to_transfer,
+    archive_directory,
+    compress=True,
+    copy_all_files=False,
+    arcname=None,
```
Comment (jan-janssen) [inline pyiron_base/project/archiving/export_archive.py]: It would be great to add type hints - now that we use co-pilot this should be easy

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Comment by samwaseda** [comment]
apparently the new case is not at all covered by unit tests.

**Comment by samwaseda** [comment]
I don't understand why it's failing. I convert it to draft for now.

**Comment by pmrv** [comment]
The error comes from `.git/objects/pack/...` files which are read-only set by git.  So I'm guessing the import tries to pack some git repository (base itself?) by accident and when the export comes around it tries to restore the same packed object files again in place, which causes the error.

**Comment by samwaseda** [comment]
I don't know why but it looks like the changes made in #1595 seem to have solved the problems.

**Pull request merged**